plugin-cache/src/yo-blacklist.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+var Q = require('q');
+var http = require('http');
+
+var log = process.env.LOGGER || console;
```

Comment:
Instead of doing this everywhere, make a utility logger and require it. That would make it easier to change the logging behavior at a later point.